src/main/java/Duke.java

```diff
@@ -1,10 +1,166 @@
+        printLine();
+        System.out.println("Bye. Hope to see you again soon!");
+        printLine();
+    }
+
+
+    public static void converse(Task[] tasks, int taskCount) {
+        Scanner sc = new Scanner(System.in);
+        String response = sc.nextLine();
+
+        boolean isNotBye = !response.equals("bye");
+
+        while (isNotBye) {
+            String[] words = response.split(" ", 2);
+            String command = words[0];
+            String detail = (words.length > 1) ? words[1] : null;
+
+            switch (command) {
+            case "list":
+                listTasks();
+                break;
+            case "mark":
+                markTask(response);
+                break;
+            case "unmark":
+                unmarkTask(response);
+                break;
+            case "todo":
+                addTask(new Todo(detail));
+                break;
+            case "deadline":
+                int byIndex = detail.indexOf("/by");
+                boolean isByPresent = byIndex != -1;
+                if (isByPresent) {
+                    String by = detail.substring(byIndex + 3).trim();
+                    String deadlineDesc = detail.substring(0, byIndex).trim();
+                    addTask(new Deadline(deadlineDesc, by));
+                } else {
+                    System.out.println("Sorry, missing /by argument...");
+                }
+                break;
+            case "event":
+                int atIndex = detail.indexOf("/at");
+                boolean isAtPresent = atIndex != -1;
+                if (isAtPresent) {
+                    String at = detail.substring(atIndex + 3).trim();
+                    String eventDesc = detail.substring(0, atIndex).trim();
+                    addTask(new Event(eventDesc, at));
+                } else {
+                    System.out.println("Sorry, missing /at argument...");
+                }
+                break;
+            default:
+                System.out.println("no such command");
+            }
+            response = sc.nextLine();
+            isNotBye = !response.equals("bye");
+        }
+    }
+
+    public static void listTasks() {
+        printLine();
+        for (int i = 0; i < taskCount; i++) {
+            System.out.print(i + 1);
+            System.out.println("." + tasks[i]);
+        }
+        printLine();
+    }
+
+    public static void addTask(Task task) {
+        printLine();
+        tasks[taskCount] = task;
+        System.out.println("Got it. I've added this task:");
+        System.out.println(task);
+        taskCount++;
+        System.out.println("Now you have " + taskCount + " task(s) in the list.");
+        printLine();
+    }
+
+    public static void markTask(String response) {
+        printLine();
+        String[] words = response.split(" ");
+        int taskIndex = Integer.parseInt(words[1]);
+        boolean isNotIndex = taskIndex > taskCount || taskIndex == 0;
+        if (isNotIndex) {
```

Comment:
Should there be a blank line to separate the variable initialisations and the if... else... logical block?